metrics/ci_integrations/test/integration/ci/sync_stage/builds/in_progress_builds_sync_stage_test.dart

```diff
@@ -52,8 +109,491 @@ void main() {
+      },
+    );
+
+    test(
+      ".call() returns an error if fetching coverage for a build fails with an error",
+      () async {
+        final inProgressBuild = BuildData(
+          buildNumber: 1,
+          buildStatus: BuildStatus.inProgress,
+          startedAt: timedOutStartedAt,
+        );
+        final inProgressBuilds = [inProgressBuild];
+        final updatedBuilds = inProgressBuilds
+            .map((build) => build.copyWith(buildStatus: BuildStatus.successful))
+            .toList();
+
+        whenFetchInProgressBuilds().thenAnswer(
+          (_) => Future.value(inProgressBuilds),
+        );
+        whenFetchBuild(inProgressBuild).thenAnswer(
+          (_) => Future.value(updatedBuilds[0]),
+        );
+        when(
+          sourceClient.fetchCoverage(updatedBuilds[0]),
+        ).thenAnswer((_) => Future.error(error));
+
+        final result = await syncStage.call(syncConfigWithCoverage);
+
+        expect(result.isError, isTrue);
+      },
+    );
+
+    test(
+      ".call() updates builds with the fetched coverage if the coverage option is enabled in the sync config",
+      () async {
+        final updatedBuilds = inProgressBuilds
+            .map((build) => build.copyWith(buildStatus: BuildStatus.successful))
+            .toList();
+
+        final coverage = Percent(0.1);
+        final updatedBuildsWithCoverage = updatedBuilds
+            .map((build) => build.copyWith(coverage: coverage))
+            .toList();
+
+        whenFetchInProgressBuilds().thenAnswer(
+          (_) => Future.value(inProgressBuilds),
+        );
+        whenFetchBuild(inProgressBuilds[0]).thenAnswer(
+          (_) => Future.value(updatedBuilds[0]),
+        );
+        whenFetchBuild(inProgressBuilds[1]).thenAnswer(
+          (_) => Future.value(updatedBuilds[1]),
+        );
+        when(
+          sourceClient.fetchCoverage(argThat(isIn(updatedBuilds))),
+        ).thenAnswer((_) => Future.value(coverage));
+
+        await syncStage.call(syncConfigWithCoverage);
+
+        verify(destinationClient.updateBuilds(
+          destinationProjectId,
+          updatedBuildsWithCoverage,
+        )).called(once);
+      },
+    );
+
+    test(
+      ".call() returns a success if updating builds succeeds",
+      () async {
+        final updatedBuilds = inProgressBuilds
+            .map((build) => build.copyWith(buildStatus: BuildStatus.successful))
+            .toList();
+
+        whenFetchInProgressBuilds().thenAnswer(
+          (_) => Future.value(inProgressBuilds),
+        );
+        whenFetchBuild(inProgressBuilds[0]).thenAnswer(
+          (_) => Future.value(updatedBuilds[0]),
+        );
+        whenFetchBuild(inProgressBuilds[1]).thenAnswer(
+          (_) => Future.value(updatedBuilds[1]),
+        );
+        when(
+          destinationClient.updateBuilds(projectId, updatedBuilds),
+        ).thenAnswer((_) => Future.value(null));
```

Comment:
```suggestion
        ).thenAnswer((_) => Future.value());
```
Please, apply this suggestion to all similar situations.